x/concentrated-liquidity/swaps.go

```diff
@@ -808,3 +808,122 @@ func edgeCaseInequalityBasedOnSwapStrategy(isZeroForOne bool, nextInitializedTic
 	}
 	return nextInitializedTickSqrtPrice.LT(computedSqrtPrice)
 }
+
+// ComputeMaxInAmtGivenMaxTicksCrossed calculates the maximum amount of the tokenInDenom that can be swapped
+// into the pool to swap through all the liquidity from the current tick through the maxTicksCrossed tick,
+// but not exceed it.
+func (k Keeper) ComputeMaxInAmtGivenMaxTicksCrossed(
+	ctx sdk.Context,
+	poolId uint64,
+	tokenInDenom string,
+	maxTicksCrossed uint64,
```

Comment:
@stackman27 I previously thought we wanted "Given X amount, how many ticks will it cross?" but since Protorev needs to bound itself from consuming too much gas and compute time, and it seems to me liquidity doesn't really impact that, but number of ticks crossed does, so it cares more about the ticks crossed.

If a method was made that given X amount how many ticks will it cross, and we put in $1k to swap in and it says it'll cross 1000 ticks, that would be too much and then the next action would be "okay how do I reduce the number of ticks crossed to something acceptable", so it feels like just querying directly for the acceptable number of ticks crossed and always using that amountIn bound is better than continuously trying in different amounts to identify what amount passes an acceptable amount of ticks.

Reply:
i see one last question, so in the current implementation how are you going to decide the maxTickCrossed value? are you going to predetermine that 50M or 100M gas can cross X amt of ticks and while generating routes for swapping make sure the tick Crossed < maxTickCrossed?

Reply:
I think the answer to this is crossing ticks should cost a fixed amt of gas so yes, your description should be accurate (but admittedly I just made this based of Skip's design requirements and treated the actual use of the API as a black box)

Reply:
@stackman27 @czarcas7ic Yes exactly. The plan is to make X number of ticks be a changeable variable (similar to pool points), and change that depending on how many CL pools we want to be able to use in backrunning.

That is, if many new CL pools launch and almost all swaps are on CL pools, then we'd want to decrease the max number of ticks crossed per pool so that we can check 2-4 routes per swap without running out of the `upperGasLimit`, if there are still more balancer swaps than CL, and our routes mainly consist of balancer pool, then we can increase X to cross more ticks per CL pool since there are less of them, etc.

Reply:
sounds good, would love to review the PR when its ready too